teamsupport/models.py

```diff
@@ -23,23 +37,102 @@ def __init__(self, client, ticket_id=None, data=None):
                 '(neither given)')
         self.id = self.TicketID
 
+    @classmethod
+    def create(cls, user_email, user_first_name, user_last_name, title,
+               description, **params):
+        # We need to associate ticket with Contact, otherwise ticket doesn't
+        # make sense. First, we try to find an existing contact.
+        contact = Contact.get(user_email)
+        if contact is None:
+            # Otherwise - create new one.
+            contact = Contact.create(
+                user_email, FirstName=user_first_name,
+                LastName=user_last_name
+            )
+
+        data = {
+            'Name': title,
+            'TicketStatusID': cls._get_ticket_status_new(),
+            'TicketTypeID': cls._get_ticket_type_support(),
+            'ContactID': contact.id
+        }
+        data.update(params)
+
+        ticket = Ticket(data=cls.get_client().create_ticket(data))
+        ticket.set_description(description)
+        return ticket
+
+    @classmethod
+    def _get_ticket_status_new(cls):
+        if cls.TICKET_STATUS_NEW is not None:
+            return cls.TICKET_STATUS_NEW
+
+        ticket_statuses = cls.get_client().get_ticket_statuses()
+        cls.TICKET_STATUS_NEW = cls._find_element_id_by_attribute_value(
+            ticket_statuses, 'TicketStatusID', 'Name', 'New')
+        return cls.TICKET_STATUS_NEW
+
+    @classmethod
+    def _get_ticket_type_support(cls):
+        if cls.TICKET_TYPE_SUPPORT is not None:
+            return cls.TICKET_TYPE_SUPPORT
+
+        ticket_types = cls.get_client().get_ticket_types()
+        cls.TICKET_TYPE_SUPPORT = cls._find_element_id_by_attribute_value(
+            ticket_types, 'TicketTypeID', 'Name', 'Support')
```

Comment:
It makes sense to make this customizable. But not now. I'm not going to get back to TS-related code few weeks until I recover. Unless we have some critical bugs.